scripts/ci/package.sh

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+
+set -e
+
+# Requires packages:
+# devscripts cmake ninja git
+
+TOP_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )/../../" && pwd )"
+
+if [ ! -d "tmp-cmake" ]; then
+mkdir tmp-cmake
+cd tmp-cmake
+wget https://cmake.org/files/v3.5/cmake-3.5.2-Linux-x86_64.tar.gz
+tar xf cmake-3.5.2-Linux-x86_64.tar.gz
+cd cmake-3.5.2-Linux-x86_64/bin/
+export PATH=`pwd`:$PATH
+else
+cd tmp-cmake/cmake-3.5.2-Linux-x86_64/bin
+export PATH=`pwd`:$PATH
+fi
+
+echo "Cloning Clang"
+
+cd $TOP_DIR/tools
+ 
+if [ ! -d "clang" ]; then
+git clone https://github.com/llvm-mirror/clang.git
+cd clang
+git checkout f37acf6b6adf1cc6c103981332f6db7f2002dcdc
+fi
+
+mkdir -p $TOP_DIR/build
+
+cd $TOP_DIR/build
+
+LLVM_CFLAGS="-I$TOP_DIR/include -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -DNDEBUG -D__NO_CTYPE_INLINE -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS"
+LLVM_CXXFLAGS="$LLVM_CFLAGS -std=gnu++11 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS  -fvisibility-inlines-hidden -fno-rtti -fexceptions -Woverloaded-virtual -Wcast-qual"
+LLVM_LDFLAGS="-L$TOP_DIR/lib -lstdc++"
+
+echo "BUILDING"
+
+LDFLAGS="$LLVM_LDFLAGS" cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_FLAGS="$LLVM_CFLAGS" -DCMAKE_CXX_FLAGS="$LLVM_CXX_FLAGS"   ../
+
+cmake --build .
+
+export PKG_DIR=$TOP_DIR/build/mono-llvm-3.9
+
+mkdir -p $PKG_DIR/usr
+
+echo "INSTALLING LLVM TO BUILD DIR"
+
+cmake -DCMAKE_INSTALL_PREFIX=$PKG_DIR/usr -P cmake_install.cmake
+
+cp -r $TOP_DIR/scripts/ci/debian $PKG_DIR/debian
+
+cd $PKG_DIR
+
+echo "MAKING DEB"
+
+dpkg-buildpackage -d -us -uc
```

Comment:
Okay, so, you're duplicating a bunch of functionality here which might (would) be easier to maintain in the "right" format.

debian/rules is a makefile, with a bunch of default targets masked by the use of the "dh" command. dpkg-buildpackage is basically running `debian/rules build` then `debian/rules binary`. The `build` target, unless you override it, basically tries to guess the build system, runs configure with default flags, then make. The `binary` target, unless you override it, basically tries to run `make install DESTDIR=debian/tmp/` or equivalent, then uses various metadata from debian/ to pack some or all of those files into a .deb

http://sources.debian.net/src/weechat/1.5-1/debian/rules/ is a good example of a good cmake-and-dh debian/rules file, which does almost everything your script does, in a more "correct" way. http://sources.debian.net/src/weechat/1.5-1/debian/rules/#L20-L21 is a great example - it overrides one of the automatic commands, dh_auto_build, adding a dependency on a cmake-generated Makefile, then specifies how to build that Makefile in a separate rule in http://sources.debian.net/src/weechat/1.5-1/debian/rules/#L7-L18